pkg/lib/cockpit-components-multi-typeahead-select.tsx

```diff
@@ -0,0 +1,390 @@
+    const focusedItem = focusedItemIndex !== null ? selectOptions[focusedItemIndex] : null;
+
+    switch (event.key) {
+      case 'Enter':
+        if (isOpen && focusedItem && focusedItem.value !== NO_RESULTS && !focusedItem.isAriaDisabled) {
+          selectOption(focusedItem?.value);
+        }
+
+        if (!isOpen) {
+          onToggle && onToggle(true);
+          setIsOpen(true);
+        }
+
+        break;
+      case 'ArrowUp':
+      case 'ArrowDown':
+        event.preventDefault();
+        handleMenuArrowKeys(event.key);
+        break;
+    }
+  };
+
+  const onToggleClick = () => {
+    onToggle && onToggle(!isOpen);
+    setIsOpen(!isOpen);
+    textInputRef?.current?.focus();
+  };
+
+  const onClearButtonClick = (_ev: React.MouseEvent) => {
+    setInputValue('');
+    onInputChange && onInputChange('');
+    resetActiveAndFocusedItem();
+    textInputRef?.current?.focus();
+  };
+
+  const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+    <MenuToggle
+      ref={toggleRef}
+      variant="typeahead"
+      aria-label="Multi select Typeahead menu toggle"
+      onClick={onToggleClick}
+      isExpanded={isOpen}
+      isDisabled={isDisabled}
+      isFullWidth
+      style={
+        {
+          width: toggleWidth
+        } as React.CSSProperties
+      }
+      {...toggleProps}
+    >
+      <TextInputGroup isPlain>
+        <TextInputGroupMain
+          value={inputValue}
+          onClick={onInputClick}
+          onChange={onTextInputChange}
+          onKeyDown={onInputKeyDown}
+          autoComplete="off"
+          innerRef={textInputRef}
+          placeholder={placeholder}
+          {...(activeItemId && { 'aria-activedescendant': activeItemId })}
+          role="combobox"
+          isExpanded={isOpen}
+          aria-controls="select-typeahead-listbox"
+        >
+            <LabelGroup numLabels={10}>
+                {selected.map((selection) => {
+                    const option = options.find((o) => o.value === selection);
+                    if (!option)
+                        return null;
+                    const { content, color } = option;
+                    function onClose(ev: React.MouseEvent<Element, MouseEvent>) {
+                        ev.stopPropagation();
+                        onRemove(selection);
+                    }
+                    return (
+                        <Label key={selection}
+                               {...(!option.isDisabled ? { onClose } : { }) }
+                               {...(color ? { color } : { }) } >
+                            {content}
+                        </Label>
+                    );
+                })}
+            </LabelGroup>
+        </TextInputGroupMain>
+        <TextInputGroupUtilities {...(!inputValue ? { style: { display: 'none' } } : {})}>
+          <Button variant="plain" onClick={onClearButtonClick} aria-label="Clear input value">
```

Comment:
Needs translation

Reply:
Oh, we also need to translate that in TypeaheadSelect.